src/ui/hooks/useWorldLoader.ts

```diff
@@ -29,24 +30,32 @@ export interface WorldLoader {
   exitWorld: () => void;
 }
 
+const getWorldGroupCall = (session: Session, world: Room) => getRoomCall(session.callHandler.calls, world.id);
+
 export function useWorldLoader(): WorldLoader {
   const { session, platform, client } = useHydrogen(true);
   const mainThread = useMainThreadContext();
-  const [matrixNetworkInterface, setMatrixNetworkInterface] = useAtom(matrixNetworkInterfaceAtom);
   const setWorld = useSetAtom(worldAtom);
 
   const exitWorldCallback = useCallback(async () => {
-    matrixNetworkInterface?.dispose();
+    provideMatrixNetworkInterface((matrixNetworkInterface) => {
+      matrixNetworkInterface?.dispose();
+    });
 
     disposeActiveMatrixRoom(mainThread);
 
     setWorld({ type: "CLOSE" });
-  }, [setWorld, matrixNetworkInterface, mainThread]);
+  }, [setWorld, mainThread]);
 
   const loadWorldCallback = useCallback(
     async (world: Room, content: Content) => {
       const roomId = world.id;
 
+      provideMatrixNetworkInterface((matrixNetworkInterface) => {
+        matrixNetworkInterface?.dispose();
+      });
+      disposeActiveMatrixRoom(mainThread);
+
```

Comment:
Did some testing and it seems that this addition of code is what broke in-world reloading. When uncommented, in-world reloading works again.

Instead, we should be calling `exitWorldCallback` outside of this file somewhere, for the case that you added these lines in for.